File: db_migrate.js
const { app } = require('electron');
const path = require('path');
const SQLite = require('better-sqlite3');
const fs = require('fs');

app.on('ready', () => {
    try {
        const userDataPath = app.getPath('userData');
        const dbPath = path.join(userDataPath, 'db', 'kiwoom.db');
        
        console.log('\n[DB 마이그레이션 시작]');
        console.log(`DB 경로: ${dbPath}`);
        
        if (!fs.existsSync(dbPath)) {
            console.error('kiwoom.db 파일을 찾을 수 없습니다.');
            return app.quit();
        }

        const db = new SQLite(dbPath);
        
        const changes1 = db.prepare("UPDATE maiis_portfolio SET status = 'HELD', last_signal = 'BUY' WHERE status = 'IMMEDIATE_BUY'").run();
        const changes2 = db.prepare("UPDATE maiis_portfolio SET status = 'WATCHING' WHERE status = 'WATCHLIST'").run();
        
        console.log(`✅ 매수 포지션 (IMMEDIATE_BUY -> HELD) 변경 완료: ${changes1.changes}건`);
        console.log(`✅ 관심 종목 (WATCHLIST -> WATCHING) 변경 완료: ${changes2.changes}건\n`);
        console.log('마이그레이션이 성공적으로 완료되었습니다!');
        
    } catch (e) {
        console.error('Migration Error:', e);
    } finally {
        app.quit();
    }
});

File: db_migrate.py
import sqlite3
import os
import sys

# 강제로 stdout 인코딩을 UTF-8로 설정하여 cp949 인코딩 에러 방지
sys.stdout.reconfigure(encoding='utf-8')

appdata = os.getenv('APPDATA')
db_path = os.path.join(appdata, 'kiwoom-trader', 'db', 'kiwoom.db')

try:
    if not os.path.exists(db_path):
        print(f"[{db_path}] DB file not found.")
        exit(1)

    conn = sqlite3.connect(db_path)
    cursor = conn.cursor()
    
    print('\n[DB Migration Start]')
    print(f'DB Path: {db_path}')
    
    cursor.execute("UPDATE maiis_portfolio SET status = 'HELD', last_signal = 'BUY' WHERE status = 'IMMEDIATE_BUY'")
    changes1 = cursor.rowcount
    
    cursor.execute("UPDATE maiis_portfolio SET status = 'WATCHING' WHERE status = 'WATCHLIST'")
    changes2 = cursor.rowcount
    
    conn.commit()
    
    print(f'- [IMMEDIATE_BUY -> HELD] Updated: {changes1}')
    print(f'- [WATCHLIST -> WATCHING] Updated: {changes2}\n')
    print('Migration Complete. Safe to restart the electron app.')
    
except Exception as e:
    print('Migration Error:', e)
finally:
    if 'conn' in locals():
        conn.close()
